src/index.js

```diff
@@ -142,7 +142,7 @@ class Josh {
     await this.readyCheck();
     if (keysOrPaths === this.all) {
       const allValues = await this.provider.getAll();
-      return this.deserializer ? allValues.map(this.deserializer) : allValues;
+      return this.deserializer ? allValues.map(value => [value[0], this.deserializer(value[1], value[0], '')]) : allValues;
```

Comment:
Here I'd rather see a proper destructuring of the row into actual variable names. So, for example, 
```js
const [ key, value ] = row;
this.deserializer(key, value)
```

Also, I'm not quite sure why we'd want to add the path, since the serializer/deserializer functions will always run on the complete value. and you're not even using it here. Please remove the path.

Reply:
I add the path here because get/set sends a path of `''` when using it and I wanted to keep things consistent. I'll go do the destructuring now though